frontend/app/components/collapsible.tsx

```diff
@@ -0,0 +1,48 @@
+import type { ComponentProps } from 'react';
+
+import { cn } from '~/utils/tw-utils';
+
+const variants = {
+  default: 'border border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300',
+};
+
+export interface CollapsableDetailsProps extends ComponentProps<'details'> {
+  id: string;
+  title: string;
+  required?: boolean;
+  label?: string;
+  variant?: keyof typeof variants;
+}
+
+export interface CollapsableSummaryProps extends ComponentProps<'details'> {
+  id: string;
+  title: string;
+  required?: boolean;
+  label?: string;
+  variant?: keyof typeof variants;
+}
+
+export function CollapsableSummary(props: CollapsableSummaryProps) {
+  const { id, title } = props;
+  const headerId = `${id}-header`;
+
+  return (
+    <summary>
+      <h2 id={headerId} className="ds-underline ml-[15px] inline font-lato text-xl font-semibold leading-8 text-blue-900 underline" dangerouslySetInnerHTML={{ __html: title }} />
+    </summary>
+  );
+}
```

Comment:
`h2` should not be use here. It will be provided by the consumer's context of the details summary if needed. Most of the time only text for to the summary is what you want.

```suggestion
export function CollapsableSummary({ children, className, ...props }: CollapsableSummaryProps) {
  return (
    <summary className={cn('text-slate-700 underline', className)} {...props}>
      <span className="ml-3">{children}</span>
    </summary>
  );
}
```

Reply:
Fixed: ead183074aeaba144439cef11bace94d8046263c

Reply:
The `CollapsableSummary` props should be spreaded and the style should be the following. Some of the classes being used are not supported by our current style system ie.: `ds-underline, w3-leftbar`
```tsx
<summary className={cn('text-slate-700 underline', className)} {...props}>
      <span className="ml-3">{children}</span>
    </summary>
```

Reply:
Fixed everything else (3f4f9eacc88244fba4ed93571cacaeb9c12b5005) but for the header color, the UI mocks ups/Figmas uses blue for collapsible header text instead of grey:
![Screenshot from 2024-03-04 12-10-41](https://github.com/DTS-STN/canadian-dental-care-plan/assets/156207329/09359547-4141-41de-93be-dc43bc6a4de4)
![Screenshot from 2024-03-04 12-11-21](https://github.com/DTS-STN/canadian-dental-care-plan/assets/156207329/2f20c22b-bcb1-4541-b2b1-579807c6cae4)

Reply:
Based on this [online tool](https://www.w3docs.com/nx/tailwindcss-closest-color) the nearest Tailwind Color would be `blue-900`. But if I used their link color `#284162` the nearest is `slate-700` as we are using for our links at the moment. Use `slate-700` to match what's being used without our project.

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/114004123/a878db9f-24a6-412f-b3a4-1b3f58667714)

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/114004123/f252d4b6-7631-4399-9963-b27817cdf51a)

Reply:
Changed to text-blue-900 as discussed: c6643f0765bde433eca607ce925740332ab9c76f.

Reply:
My bad, FIxed: 3cdd30bfd16fa5688eed4254581cef52a7539416

Reply:
`details` has been fixed but not `summary`

Reply:
Fixed: 8399842d80553b8946baababc0cc0196562355c4